tools/edbconnect/main/main.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"bufio"
+	"database/sql"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ten-protocol/go-ten/go/enclave/storage/init/edgelessdb"
+	"github.com/ten-protocol/go-ten/integration/common/testlog"
+)
+
+func main() {
+	fmt.Println("Retrieving Edgeless DB credentials...")
+	creds, found, err := edgelessdb.LoadCredentialsFromFile()
+	if err != nil {
+		fmt.Println("Error loading credentials from file:", err)
+		panic(err)
+	}
+	if !found {
+		panic("No existing EDB credentials found.")
+	}
+	fmt.Println("Found existing EDB credentials. Creating TLS config...")
+	cfg, err := edgelessdb.CreateTLSCfg(creds)
+	if err != nil {
+		fmt.Println("Error creating TLS config from credentials:", err)
+		panic(err)
+	}
+	fmt.Println("TLS config created. Connecting to Edgeless DB...")
+	testlog.SetupSysOut()
+	db, err := edgelessdb.ConnectToEdgelessDB("obscuronode-edgelessdb", cfg, testlog.Logger())
+	if err != nil {
+		fmt.Println("Error connecting to Edgeless DB:", err)
+		panic(err)
+	}
+	fmt.Println("Connected to Edgeless DB.")
+
+	startREPL(db)
+
+	err = db.Close()
+	if err != nil {
+		fmt.Println("Error closing Edgeless DB connection:", err)
+		panic(err)
+	}
+}
+
+// Starts a loop that reads user input and runs queries against the Edgeless DB until user types "exit"
+func startREPL(db *sql.DB) {
```

Comment:
Have we tested how special characters behave here, considering its coming from stdio? I imagine since its in app the terminal won't be doing any ad hoc bs, but ...

Reply:
If you press like keys like up and ctrl on the keyboard it does create some garbage characters I think, let me have a look see if I can improve that. Didn't see any weird rendering with the query responses though (although it's not very pretty, nicer formatting and option to dump response to CSV would be useful upgrades).